mkosi/__init__.py

```diff
@@ -2255,3 +2270,24 @@ def run_verb(args: MkosiArgs, presets: Sequence[MkosiConfig]) -> None:
 
             if args.verb == Verb.serve:
                 run_serve(last)
+
+
+def normalize_mtime(root: Path, source_date_epoch: Optional[int]) -> None:
+    if source_date_epoch is None:
+        return
+    with complete_step("Setting mtime to SOURCE_DATE_EPOCH"):
+        os.utime(root, (source_date_epoch, source_date_epoch), follow_symlinks=False)
+        for p in root.rglob("*"):
+            os.utime(p, (source_date_epoch, source_date_epoch), follow_symlinks=False)
+
+
+def normalize_mtime_path(root: Path, target: Path, source_date_epoch: Optional[int]) -> None:
+    if source_date_epoch is None:
+        return
+    assert root in target.parents
```

Comment:
`asserts` are problematic in Python, since they are easy to lose (e.g. setting the environment variable `PYTHONOPTIMIZE=1`). They're ok for type refinements, but to catch an error, make it
```suggestion
    if  root not in target.parents:
        raise WhichEverExceptionYouThinkSensible("some helpful message")
```
and put empty lines around it for readability.